File: examples/notAction/notAction.py
from iam_floyd import statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')

s = (
    # doc-start
    statement.S3() \
        .allow() \
        .not_actions() \
        .to_delete_bucket() \
        .on_bucket('example-bucket')
    # doc-end
)

all = [s]
out(all)
deploy(all)

File: examples/notAction/notAction.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Statement } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Statement.S3()
      .allow()
      .notAction()
      .toDeleteBucket()
      .onBucket('example-bucket')
    // doc-end
  );
}
const s = [getStatement()];
out(s);
deploy(s);
